File: src/python/_1137_N-th_Tribonacci_Number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0:
            return 0

        if n == 1 or n == 2:
            return 1

        i = 0
        j = 1
        k = 1

        for _ in range(n - 2):
            i, j, k = j, k, i + j + k

        return k

File: src/c/_1137_N-th_Tribonacci_Number.c
int tribonacci(int n)
{
    if (n == 0)
        return 0;

    if (n == 1 || n == 2)
        return 1;

    int a = 0, b = 1, c = 1, val;

    for (int i = 2; i < n; i++)
    {
        val = a + b + c;
        a = b;
        b = c;
        c = val;
    }

    return c;
}
